app/components/transUnit/TransUnitService.js

```diff
@@ -170,7 +198,26 @@
        updateSaveButton);
 
     function modifyTranslationText(phrase, newText) {
-      phrase.newTranslation = newText;
+      if (phrase.plural) {
+        var transUnitCtrl = controllerList[phrase.id];
+        phrase.newTranslations[transUnitCtrl.focusedTranslationIndex] = newText;
+      } else {
+        phrase.newTranslation = newText;
+      }
+      EventService.emitEvent(EventService.EVENT.TRANSLATION_TEXT_MODIFIED,
+        phrase);
+      EventService.emitEvent(EventService.EVENT.FOCUS_TRANSLATION,
+        phrase);
+    }
+
+    function modifyTranslationsText(phrase, newTexts) {
```

Comment:
I think the best way to do it is to convert the singular 'content' to an array as soon as we get it from the REST call, and convert it back just before it is sent to the server for a save. This would mean that most of the code can just assume there is an array and does not need to branch on plural/singular. In the few places that we need different display or logic, we can just use the `.isPlural` property to branch on.